electron/main.js

```diff
@@ -0,0 +1,89 @@
+import net from 'net';
+import { join } from 'path';
+import { app, BrowserWindow, utilityProcess, dialog } from 'electron';
+
+// Vite default dev & production ports
+const hollamaPort = app.isPackaged ? '4173' : '5173';
+const HOLLAMA_HOST = '127.0.0.1';
+
+function createWindow() {
+	const mainWindow = new BrowserWindow({
+		width: 1280,
+		height: 800,
+		minWidth: 400,
+		minHeight: 640
+	});
+
+	mainWindow.menuBarVisible = false; // Windows: hides the menu bar
+	mainWindow.loadURL(`http://${HOLLAMA_HOST}:${hollamaPort}`);
+}
+
+function checkServerAvailability(port) {
+	const MAX_RETRIES = 10;
+	const RETRY_INTERVAL_IN_MS = 1000;
+
+	return new Promise((resolve, reject) => {
+		let retries = 0;
+
+		function tryConnection() {
+			const socket = new net.Socket();
+
+			const onError = () => {
+				socket.destroy();
+
+				if (retries >= MAX_RETRIES) {
+					reject(new Error(`Couldn't connect to Hollama server after ${MAX_RETRIES} attempts`));
+				} else {
+					retries++;
+					setTimeout(tryConnection, RETRY_INTERVAL_IN_MS);
+				}
+			};
+
+			socket.setTimeout(1000);
+			socket.once('error', onError);
+			socket.once('timeout', onError);
+
+			socket.connect(port, HOLLAMA_HOST, () => {
+				socket.destroy();
+				resolve();
+			});
+		}
+
+		tryConnection();
+	});
+}
```

Comment:
We use this function to check if the server is ready once per second, up-to 10 times.